cpp/src/io/csv/reader_impl.cu

```diff
@@ -756,24 +760,40 @@ table_with_metadata read_csv(cudf::io::datasource* source,
   }
 
   // User can specify which columns should be parsed
-  if (!reader_opts.get_use_cols_indexes().empty() || !reader_opts.get_use_cols_names().empty()) {
+  auto const unique_use_cols_names = std::unordered_set<std::string>(
+    reader_opts.get_use_cols_names().cbegin(), reader_opts.get_use_cols_names().cend());
+  auto const is_column_selection_used =
+    not unique_use_cols_names.empty() or not unique_use_cols_indexes.empty();
+
+  if (is_column_selection_used) {
     std::fill(column_flags.begin(), column_flags.end(), column_parse::disabled);
+    num_active_columns = 0;
```

Comment:
What makes a column "active"? I'm not sure why this is set to 0 here.

Reply:
Active means that it's selected and included in the output dataframe. If we don't do column selection, all columns are active (we read all columns). If column selection is used, we clear active columns (i.e. `fill` in the line above) and then re-set flags for the selected columns. `num_active_columns` is updated to match `column_flags`.